### PR TITLE
Add public accessors to AppData and AppDataSummary classes

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Adds public accessors to AppData and AppDataSummary, for use by 3rd party developers and internal API (denoted with marker annotation).

Note: the intention is for further work to be carried out in a separate PR (e.g. migrating legacy fields to metadata, extracting the data capture to a separate class, etc)

### Changed

Converted fields/methods to properties (getter/setter backed with a field) in AppData and AppDataSummary.

## Tests

Adapted unit tests for JSON serialisation of each class, and added a test for the default value + override value of each accessor.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

**Comment by fractalwrench** [comment]
@martin308 I made a start with these two classes. Let me know if you're happy with the general approach to tests/docs, as I can split things apart into separate PRs if it makes easier for review.

My intention at the minute is to split out further work in separate PRs (e.g. adding accessors on the client object, refactoring the data capture)

File: sdk/src/main/java/com/bugsnag/android/AppData.java [hunk 1]
```diff
@@ -34,7 +41,12 @@
         this.appContext = appContext;
         this.sessionTracker = sessionTracker;
         appName = getAppName(appContext);
-        packageName = getPackageName(appContext);
+
+        packageName = calculatePackageName(appContext);
+        buildUuid = config.getBuildUUID();
+        durationMs = getDurationMs();
+        foregroundMs = sessionTracker.getDurationInForegroundMs(System.currentTimeMillis());
+        inForeground = sessionTracker.isInForeground();
```
Comment (martin308) [inline sdk/src/main/java/com/bugsnag/android/AppData.java]: is there an issue with pre calculating these fields that they won't be true at the time we read the data?

**Comment by martin308** [comment]
@fractalwrench thanks for the preview! I added one question around pre calculating some of the fields but apart from that it looks good to me

File: sdk/src/main/java/com/bugsnag/android/AppData.java (hunk 1 above)
Comment (fractalwrench) [reply sdk/src/main/java/com/bugsnag/android/AppData.java]: For the purposes of Unity, I don't think so. I believe the intention is that the Unity layer will call something like: `client.generateAppData()` which will instantiate a new object each time.

For Android, I'd anticipate that these values could be slightly different from previous versions, as they should now be calculated at the time of report capture, rather than at the point of serialisation. I think on average it would be different by the order of milliseconds, and worst case, seconds. We can add a general note about this - the values will be more correct IMO.

File: example/build.gradle [hunk 2]
```diff
@@ -28,7 +28,7 @@ android {
         targetSdkVersion Integer.parseInt(project.ANDROID_TARGET_SDK_VERSION)
 
         manifestPlaceholders = [
-            bugsnagApiKey                    : "your-api-key",
+            bugsnagApiKey                    : "572dd1154e6c5b291e5d1b42e506ae34",
```
Comment (martin308) [inline example/build.gradle]: Should this be checked in?

File: sdk/src/main/java/com/bugsnag/android/AppData.java [hunk 3]
```diff
@@ -34,31 +41,36 @@
...
     }
 
+    // TODO migrate metadata fields to separate class
+    void addAppMetaData(MetaData metaData) {
+        metaData.addToTab("app", "name", appName);
+        metaData.addToTab("app", "packageName", packageName);
+        metaData.addToTab("app", "versionName", getVersionName());
+        metaData.addToTab("app", "activeScreen", getActiveScreenClass());
+        metaData.addToTab("app", "memoryUsage", getMemoryUsage());
+        metaData.addToTab("app", "lowMemory", isLowMemory(appContext));
```
Comment (martin308) [inline sdk/src/main/java/com/bugsnag/android/AppData.java]: How would the unity notifier access these values?

Comment (fractalwrench) [reply sdk/src/main/java/com/bugsnag/android/AppData.java]: Comment addressed in #333
